rbd/rbd.go

```diff
@@ -831,7 +831,7 @@ func (image *Image) WriteSame(ofs, n uint64, data []byte, flags rados.OpFlags) (
 
 	ret := C.rbd_writesame(image.image,
 		C.uint64_t(ofs),
-		C.uint64_t(n),
+		C.size_t(n),
```

Comment:
That seems to be the correct fix.

https://github.com/ceph/ceph/blob/67fb1b157c23b71c8f8e3c7e61c122996f97e49b/src/include/rbd/librbd.h#L1151-L1153